packages/@n8n/api-types/src/dto/folders/__tests__/list-folder-query.dto.test.ts

```diff
@@ -0,0 +1,161 @@
+import { ListFolderQueryDto } from '../list-folder-query.dto';
+
+describe('ListFolderQueryDto', () => {
+	describe('Valid requests', () => {
+		test.each([
+			{
+				name: 'empty object (no filters)',
+				request: {},
+			},
+			{
+				name: 'valid filter',
+				request: {
+					filter: '{"name":"test"}',
+				},
+			},
+			{
+				name: 'filter with parentFolderId',
+				request: {
+					filter: '{"parentFolderId":"abc123"}',
+				},
+			},
+			{
+				name: 'filter with name and parentFolderId',
+				request: {
+					filter: '{"name":"test","parentFolderId":"abc123"}',
+				},
+			},
+			{
+				name: 'filter with tags array',
+				request: {
+					filter: '{"tags":["important","archived"]}',
+				},
+			},
+			{
+				name: 'filter with empty tags array',
+				request: {
+					filter: '{"tags":[]}',
+				},
+			},
+			{
+				name: 'filter with all properties',
+				request: {
+					filter: '{"name":"test","parentFolderId":"abc123","tags":["important"]}',
+				},
+			},
+			{
+				name: 'valid select',
+				request: {
+					select: '["id","name"]',
+				},
+			},
+			{
+				name: 'valid sortBy',
+				request: {
+					sortBy: 'name:asc',
+				},
+			},
+			{
+				name: 'valid skip and take',
+				request: {
+					skip: '0',
+					take: '10',
+				},
+			},
+			{
+				name: 'full query parameters',
+				request: {
+					filter: '{"name":"test","tags":["important"]}',
+					select: '["id","name","createdAt","tags"]',
+					skip: '0',
+					take: '10',
+					sortBy: 'createdAt:desc',
+				},
+			},
+		])('should validate $name', ({ request }) => {
+			const result = ListFolderQueryDto.safeParse(request);
+			expect(result.success).toBe(true);
```

Comment:
We don't have any tests here testing that what these DTOs produce are what we expect, and as there's some amount of logic in there I feel like we maybe should have some.